docs/src/guides/rust_to_wasm.md

```diff
@@ -0,0 +1,176 @@
+# Compiling Rust To WebAssembly
+
+This chapter will walk you through compiling a Rust crate to a WebAssembly (Wasm) module
+in binary (i.e. `.wasm`) form. The Miden compiler has a frontend which can take such
+modules and compile them on to Miden Assembly, which will be covered in the next chaper.
+
+## Setup
+
+First, let's set up a simple Rust project that contains an implementation of the Fibonacci
+function (I know, its overdone, but we're trying to keep things as simple as possible to 
+make it easier to show the results at each step, so bear with me):
+
+Start by creating a new library crate:
+
+    $ cargo new --lib wasm-fib && cd wasm-fib
+
+
+To compile to WebAssembly, you must have the appropriate Rust toolchain installed, and we
+will also need additional Cargo nightly features to build for Miden, so let's add a toolchain
+file to our project root so that `rustup` and `cargo` will know what we need, and use them by
+default:
+
+    $ cat <<EOF > rust-toolchain.toml
+    [toolchain]
+    channel = "nightly"
+    targets = "wasm32-unknown-unknown"
+    EOF
+
+Next, edit the `Cargo.toml` file as follows:
+
+```toml
+[package]
+name = "wasm-fib"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+# Build this crate as a self-contained, C-style dynamic library
+# This is required to emit the proper Wasm module type
+crate-type = ["cdylib"]
+
+[dependencies]
+# Use a tiny allocator in place of the default one, if we want
+# to make use of types in the `alloc` crate, e.g. String. We
+# don't need that now, but its good information to have in hand.
+#wee_alloc = "0.4"
```

Comment:
Judging by the issues https://github.com/rustwasm/wee_alloc/issues the project has critical bugs and does not seem to be maintained anymore.

Reply:
It's not a great allocator, but it _is_ extremely small, the issues people have with it stem from the fact that it doesn't do any complex freelist management, so it "leaks" memory, but for a great many Wasm applications this is perfectly suitable since they are not generally long-running applications.

That said, I will add a note about the tradeoffs with using it, just so people realize the potential downsides - but I would still recommend it in a variety of single-shot style programs. One could even use a stupid-simple bump allocator, which would be even smaller, but useful for a number of use cases as well. I could see it being beneficial for Miden to write a custom allocator that makes better tradeoffs for a ZK machine in the future, but something like dlmalloc is overkill for the kinds of simple applications people are going to be targeting Miden with.